### PR TITLE
Allow OpUndef appearing in the type and global variable section.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

**Comment by dneto0** [comment]
Please also make a test case using OpUndef inside a basic block inside a function.

**Comment by antiagainst** [comment]
Added a new test.

**Comment by Qining** [comment]
+2

**Comment by Qining** [comment]
Hmm, I'd prefer to rename the function "RoundtripCheck" to something like AssembleThenDisassemble.

File: source/opt/ir_loader.cpp [hunk 1]
```diff
@@ -87,7 +87,7 @@ void IrLoader::AddInstruction(const spv_parsed_instruction_t* inst) {
         module_->AddType(std::move(spv_inst));
       } else if (IsConstantInst(opcode)) {
         module_->AddConstant(std::move(spv_inst));
-      } else if (opcode == SpvOpVariable) {
+      } else if (opcode == SpvOpVariable || opcode == SpvOpUndef) {
```
Comment (dneto0) [inline source/opt/ir_loader.cpp]: But an Undef is not a GlobalVariable.
Maybe the method is misnamed?

Comment (antiagainst) [inline source/opt/ir_loader.cpp]: What about only having `AddType` and `AddGlobalValue`? We can use the latter to add constants, variables, & `OpUndef`s.

Comment (dneto0) [inline source/opt/ir_loader.cpp]: Sounds good to me.

Comment (antiagainst) [inline source/opt/ir_loader.cpp]: Done.

**Comment by dneto0** [comment]
+2

**Pull request merged**